Controllers/CategoryController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using FiveDevsShop.Data;
+using FiveDevsShop.Models;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FiveDevsShop.Controllers
+{
+    public class CategoryController : Controller
+    {
+        private readonly ApplicationDbContext db;
+
+        public CategoryController(ApplicationDbContext db)
+        {
+            this.db = db;
+        }
+
+        public IActionResult GetCategoryAndSubcategories(int? id)
+        {
+            FiveDevsShop.Models.Category category;
+            List<Category> subCategories = new List<Category>();
+
+            //The first element in subcategories list is a category and the later ones are its subcategories.
+            if(id == null)
+            {
+                subCategories = db.Category.Where(c => c.Parent_id == null).ToList();
+                subCategories.Insert(0, null);
```

Comment:
kodėl reikia insertinti null ?

Reply:
Pirmasis sąrašo elementas saugo tėvinę kategoriją, o likę - vaikines kategorijas. Kadangi šis atvejis apima 1 lygio kategorijų gavimą, tai yra įterpiamas null, kadangi 1 lygis tėvinių kategorijų neturi.